src/printer.js

```diff
@@ -1,4 +1,46 @@
 // https://prettier.io/docs/en/plugins.html#printers
-const genericPrint = (path, options) => options.originalText;
+const { concat, line } = require('prettier').doc.builders;
+
+const space = ' ';
+
+const genericPrint = (path, options) => {
+  const node = path.getValue();
+  const lines = options.originalText.split('\n');
+  const parts = [];
+
+  switch (node.type) {
+    case 'SourceUnit':
+      node.children.forEach(child => {
```

Comment:
To avoid this nesting, you can move the `PragmaDirective` case into the parent `switch` statement, then call `path.map` in the `SourceUnit` printer as I suggested earlier.